test/ch/epfl/leb/sass/simulator/loggers/FrameLoggerTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2017 Laboratory of Experimental Biophysics
+ * Ecole Polytechnique Federale de Lausanne
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package ch.epfl.leb.sass.simulator.loggers;
+
+import ch.epfl.leb.sass.simulator.loggers.StateLogger;
+import java.io.File;
+import java.io.IOException;
+import java.io.FileReader;
+import java.io.BufferedReader;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+import static org.junit.Assert.*;
+
+/**
+ * Logs all state transitions from a simulation.
+ * @author Kyle M. Douglass
+ */
+public class StateLoggerTest { 
+    private StateLogger logger = null;
+    
+    @Rule
+    public TemporaryFolder tempDir = new TemporaryFolder();
+    
+    public StateLoggerTest() {
+    }
+    
+    @Before
+    public void setUp () {
+        logger = StateLogger.getInstance();
+    }
+    
+    /**
+     * Test of logStateTransition method, of class StateLogger.
+     */
+    @Test
+    public void testLogStateTransition() {
+        System.out.println("logStateTransition");
+        int id = 1;
+        double timeElapsed = 10.0;
+        int initialState = 0;
+        int nextState = 1;
+        logger.setPerformLogging(true);
+        
+        // Method call to test is here.
+        logger.logStateTransition(id, timeElapsed, initialState, nextState);
+        
+        int actualId = logger.getIds().get(0);
+        double actualTimeElapsed = logger.getElapsedTimes().get(0);
+        int actualInitialState = logger.getInitialStates().get(0);
+        int actualNextState = logger.getNextStates().get(0);
+        assertEquals(id, actualId);
+        assertEquals(timeElapsed, actualTimeElapsed, 0.001);
+        assertEquals(initialState, actualInitialState);
+        assertEquals(nextState, actualNextState);
+        
+    }
+    
+    /**
+     * Test for resetting the logger to its initial state.
+     */
+    @Test
+    public void testReset() throws IOException {
```

Comment:
This will require some custom code that is specific to how FrameLogger works.

Reply:
I guess I didn't commit properly the first time. Hopefully this time everything should be up to date. Sorry about that.